src/System.Management.Automation/FormatAndOutput/common/Utilities/Mshexpression.cs

```diff
@@ -159,27 +160,43 @@ public List<PSPropertyExpression> ResolveNames(PSObject target, bool expand)
 
             // If the object passed in is a hashtable, then turn it into a PSCustomObject so
             // that property expressions can work on it.
-            target = IfHashtableWrapAsPSCustomObject(target);
+            var wrappedTarget = IfHashtableWrapAsPSCustomObject(target, out bool wasHashtable);
 
             // we have a string value
-            IEnumerable<PSMemberInfo> members = null;
+            IEnumerable<PSMemberInfo> members;
             if (HasWildCardCharacters)
             {
                 // get the members first: this will expand the globbing on each parameter
-                members = target.Members.Match(_stringValue,
-                                            PSMemberTypes.Properties | PSMemberTypes.PropertySet | PSMemberTypes.Dynamic);
+                members = wrappedTarget.Members.Match(
+                    _stringValue,
+                    PSMemberTypes.Properties | PSMemberTypes.PropertySet | PSMemberTypes.Dynamic);
+
+                // if target was a hashtable and no result is found from the keys, then use property value if available
+                if (wasHashtable && !members.Any())
+                {
+                    members = target.Members.Match(
+                        _stringValue,
+                        PSMemberTypes.Properties | PSMemberTypes.PropertySet | PSMemberTypes.Dynamic);
```

Comment:
We evaluate this again. Maybe put this in a variable before line 172?

Reply:
This is evaluated against a different target object in the prior case.

Reply:
I mean `PSMemberTypes.Properties | PSMemberTypes.PropertySet | PSMemberTypes.Dynamic`